src/more-infos/more-info-alarm_control_panel.js

```diff
@@ -34,17 +34,44 @@ export default new Polymer({
       type: Boolean,
       value: false,
     },
+    code_input_visible: {
+      type: Boolean,
+      value: false,
+    },
+    code_input_enabled: {
+      type: Boolean,
+      value: false,
+    },
+    code_format: {
+      type: String,
+      value: '',
+    },
+    code_valid: {
+      type: Boolean,
+      value: false,
+    },
   },
-  enteredCodeChanged(ev) {
-    this.entered_code = ev.target.value;
+  validate_code(code) {
+    const re = new RegExp(this.code_format);
```

Comment:
I don't really understand what you mean. Should I have a last checked enteredCode and don't check with regex if the enteredCode hasn't changed?